common/include/pcl/point_cloud.h

```diff
@@ -468,7 +547,7 @@ namespace pcl
         points.resize (n);
         if (width * height != n)
         {
-          width = static_cast<std::uint32_t> (n);
+          width = static_cast<index_t> (n);
```

Comment:
Which methods do you mean? In the methods below this, we're only casting `size()` right? Since in the future `size()` will return `index_t`, I reckon we should be able to simply remove the cast at that point.

Reply:
I was referring to parameter `size_t n` of `PointCloud::resize(size_t)`. Currently it takes a size_t, however width, at best can take `index_t`. Hence, my suggestion was to add
```cpp
PointCloud::resize(index_t)
```
and deprecated
```cpp
PointCloud::resize(size_t)
```

Reply:
Sorry my comment wasn't very clear. What I was trying to ask was, what are the other methods 'below' `resize (std::size_t n)` that you mentioned in this comment?
> Following the brief Discord exchange, make the same change for the methods below this one.

Reply:
```cpp
inline const PointT& at (int column, int row) const;
inline PointT& at (int column, int row);
inline const PointT& operator () (std::size_t column, std::size_t row) const;
inline PointT& operator () (std::size_t column, std::size_t row)

// less important required to to enforce consistency
inline void reserve (std::size_t n) { points.reserve (n); }
inline const PointT& operator[] (std::size_t n) const { return (points[n]); }
inline PointT& operator[] (std::size_t n) { return (points[n]); }
inline const PointT& at (std::size_t n) const { return (points.at (n)); }
inline PointT& at (std::size_t n) { return (points.at (n)); }
inline void insert (iterator position, std::size_t n, const PointT& pt)
```